server/queries/db.js

```diff
@@ -17,23 +18,98 @@ sequelize.sync({ alter: true }).then(() => {
     console.log("Database connected.");
 });
 
-module.exports.createUser = username => {
+module.exports.validatePassword = (username, password) => {
+    return new Promise((resolve, reject) => {
+        User.findOne({
+            where: {
+                username: username
+            }
+        })
+        .then(user => {
+            if (user) {
+                bcrypt.compare(password, user.dataValues.password, (err, res) => {                    
+                    if (res) {
+                        resolve(user);
+                    } else {
+                        reject({
+                            code: 404,
```

Comment:
`/auth/login` doesn't appear to use these codes

Reply:
are you intending to describe an HTTP status code?

404 would be inappropriate. the resource `/auth/login` was found.

401 Unauthorized would be appropriate for a failed authentication attempt.

Reply:
worth considering 200 OK as in "there were no login errors. you used the endpoint correctly." but the body could say "`{success: false}`"

Reply:
why would the success be false?